src/compiler/checker.ts

```diff
@@ -14165,6 +14177,26 @@ namespace ts {
             }
         }
 
+        function isSourceFileADefinitionFile(node: Node): boolean {
+            return getSourceFileOfNode(node).fileName.indexOf(".d.ts") > 0;
+        }
+
+        function checkUnusedIdentifiers(node: FunctionDeclaration | MethodDeclaration | ConstructorDeclaration): void {
+            if (!isSourceFileADefinitionFile(node)) {

```

Comment:
i would also check for node.body to avoid reporting errors for function overloads.